src/transport_performance/utils/raster.py

```diff
@@ -0,0 +1,167 @@
+
+    # get tif files in directory, ensure some exist and select subset via regex
+    tif_filepaths = glob.glob(f"{input_dir}/*.tif")
+    if len(tif_filepaths) == 0:
+        raise FileNotFoundError(f"No `*.tif` files found in {input_dir}")
+
+    # apply regex and ensure tif files exist after applying it
+    # raise a unique FileNotFoundError to aid debugging
+    if subset_regex is not None:
+        tif_filepaths = [
+            fpath for fpath in tif_filepaths if re.search(subset_regex, fpath)
+        ]
+    if len(tif_filepaths) == 0:
+        raise FileNotFoundError(
+            f"No `*.tif` files found in {input_dir} after applying regex "
+            f"'{subset_regex}'."
+        )
+
+    # build a list of input rioxarrays to be merged
+    arrays = []
+    for tif_filepath in sorted(tif_filepaths):
+        arrays.append(rioxarray.open_rasterio(tif_filepath, masked=True))
+
+    # merge the datasets together
+    xds_merged = merge_arrays(arrays)
+
+    # make output_dir if it does not exist
+    if not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+
+    # create full filepath for merged tif file and write to disk
+    MERGED_DIR = os.path.join(output_dir, output_filename)
+    xds_merged.rio.to_raster(MERGED_DIR)
+
+    # get boundaries of inputs and output raster
+    bounds = {
+        "inputs": [array.rio.bounds() for array in arrays],
+        "output": [xds_merged.rio.bounds()],
+    }
+
+    return bounds
+
+
+def sum_resample_file(
+    input_filepath: str,
+    output_filepath: str,
+    resample_factor: int = 2,
+) -> None:
+    """Resample raster file (change grid resolution) by summing.
+
+    Takes an input raster file, resamples to a different grid resolution by
+    summing constituent cells, and writes the resampled raster to file.
+
+    Parameters
+    ----------
+    input_filepath : str
+        Input filpath of GeoTIFF file
+    output_filepath : str
+        Output filepath for resampled GeoTIFF file
+    resample_factor : int, optional
+        Factor to resample input raster by, by default 2 which means the
+        resolution will be decreased by 2 (e.g., input=100x100m then output=200
+        x200m)
+
+    Raises
+    ------
+    FileNotFoundError
+        Unable to find `input_filepath`
+
+    """
+    # defend against case where the provided input dir does not exist
+    if not os.path.exists(input_filepath):
+        raise FileNotFoundError(f"{input_filepath} can not be found")
+
+    xds = rioxarray.open_rasterio(input_filepath, masked=True)
+
+    # resample based on scaling factor and using sum resampling
+    xds_resampled = xds.rio.reproject(
+        xds.rio.crs,
+        resolution=tuple(
+            res * resample_factor for res in xds.rio.resolution()
+        ),
+        resampling=Resampling.sum,
+    )
+
+    # make output_filepath's directory if it does not exist
+    if not os.path.exists(os.path.dirname(output_filepath)):
+        os.mkdir(output_filepath)
```

Comment:
Would be worth seeing if [any of the utility functions I’ve written](https://github.com/datasciencecampus/transport-network-performance/blob/9cb601de0461002f71c39e6a79830db70d4b4131/src/transport_performance/utils/defence.py) (not yet merged to main) would suffice. Or refactoring them. Example implementation below:

```
from transport_performance.utils.defence import _check_parent_dir_exists
…
```
Then in `raster.merge_raster_files`:
```
_check_parent_dir_exists(pth=output_filepath, param_nm=“output_filepath”, create=True)
```
If this meets your behavioural reqs, this function has already been tested so you should be good to go.